testing/integration/fleetserver_fips_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+)
+
+const cloudAgentPolicyID = "policy-elastic-agent-on-cloud"
+
+// TestFIPSAgentConnectingToFIPSFleetServerInECHFRH ensures that a FIPS-capable Elastic Agent
+// running in an ECH FRH (FedRamp High) environment is able to successfully connect to its
+// own local Fleet Server instance (which, by definition should also be FIPS-capable and
+// running in the ECH FRH environment).
+func TestFIPSAgentConnectingToFIPSFleetServerInECHFRH(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Group: Fleet,
+		Stack: &define.Stack{},
+		OS: []define.OS{
+			{Type: define.Linux},
+		},
+		Sudo:  false,
+		Local: true,
+
+		// Ensures the test will run in a FIPS-configured environment against a
+		// deployment in ECH that's running a FIPS-capable integrations server.
+		FIPS: true,
+	})
+
+	// Further, the Fleet Server must be configured with FIPS-compliant TLS (TLSv1.2
+	// and TLSv1.3 and appropriate ciphers).
```

Comment:
@michel-laterman is this something that will be taken care of when provisioning the ESS FRH region deployment?

Reply:
Our containers run with a complaint openssl installation